### PR TITLE
✨ Allow logged in user to view list of created articles

### PR DESCRIPTION
**Pull request by Esseme**
A logged in user is now able to view the list of articles that they have
created.

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class ArticlesController < ApplicationController
   before_action :set_article, only: [:show, :edit, :update]
 
+  def index
+    @articles = current_user.articles.all
```
Comment (purinkle) [inline app/controllers/articles_controller.rb:5]: **suggestion (non-blocking):** Add some ordering.

What order do we expect to display these articles? There is an implied ordering, but it pays to make these things explicit.

File: spec/system/user_articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+RSpec.describe "User views their articles" do
+  it "displays a list of articles created by the user" do
+    user = create(:user)
+    visit root_path(as: user)
```
Comment (purinkle) [inline spec/system/user_articles_spec.rb:6]: **todo:** Drop this visit.

There is no need for this visit. All it does is add an unnecessary page visit to your tests and slow them down. You can drop it and skip that step.

File: spec/system/user_articles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    3.times { create(:article, user: user) }
+    visit articles_path
+
+    user.articles.each do |article|
+      expect(page).to have_content(article.title)
+      expect(page).to have_content(article.content)
+    end
```
Comment (purinkle) [inline spec/system/user_articles_spec.rb:13]: **suggestion (blocking):** Rewrite this spec.

How do you feel about the following?

```ruby
create :article, user: user, title: "TEST_ARTICLE_1"
create :article, user: user, title: "TEST_ARTICLE_2"
create :article, title: "TEST_ARTICLE_3"

visit articles_path

expect(page).to have_content "TEST_ARTICLE_1"
expect(page).to have_content "TEST_ARTICLE_2"
expect(page).not_to have_content "TEST_ARTICLE_3"
```

It pays to be more explicit with these things, and you should avoid loops in your specs. They only make the reader try and figure out what is happening. Instead, we should make it as straightforward as possible.

Also, you want to test what we hide from the user. I added a test to ensure they only see articles that belong to them.

What do you think?

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (Esseme) [reply app/controllers/articles_controller.rb:5]: Addressed in PR #49

File: spec/system/user_articles_spec.rb (hunk 3 above)
Comment (Esseme) [reply spec/system/user_articles_spec.rb:13]: Addressed in PR #49

**Pull request merged**